vms/platformvm/state/state.go

```diff
@@ -834,62 +840,43 @@ func (s *state) DeleteExpiry(entry ExpiryEntry) {
 	s.expiryDiff.DeleteExpiry(entry)
 }
 
-func (s *state) GetCurrentValidatorSet(ctx context.Context, subnetID ids.ID) (map[ids.ID]*validators.GetCurrentValidatorOutput, uint64, error) {
-	result := make(map[ids.ID]*validators.GetCurrentValidatorOutput)
+func (s *state) GetCurrentValidators(ctx context.Context, subnetID ids.ID) ([]*Staker, []L1Validator, uint64, error) {
 	// First add the current validators (non-L1)
-	for _, staker := range s.currentStakers.validators[subnetID] {
+	legacyBaseStakers := s.currentStakers.validators[subnetID]
+	legacyStakers := make([]*Staker, 0, len(legacyBaseStakers))
+	for _, staker := range legacyBaseStakers {
 		if err := ctx.Err(); err != nil {
-			return nil, 0, err
-		}
-		validator := staker.validator
-		result[validator.TxID] = &validators.GetCurrentValidatorOutput{
-			ValidationID:  validator.TxID,
-			NodeID:        validator.NodeID,
-			PublicKey:     validator.PublicKey,
-			Weight:        validator.Weight,
-			StartTime:     uint64(validator.StartTime.Unix()),
-			MinNonce:      0,
-			IsActive:      true,
-			IsL1Validator: false,
+			return nil, nil, 0, err
```

Comment:
Remove that if condition, appending will take near-zero time, there is no point checking the context.